digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -172,27 +171,40 @@ where
 
 pub async fn get_related_for_assets(
     conn: &impl ConnectionTrait,
-    assets: Vec<(asset::Model, Option<asset_data::Model>)>,
+    assets: Vec<asset::Model>,
```

Comment:
Long story short: previously the join between `asset` and `asset_data` was done outside of `get_related_for_assets` fn, but I've moved it inside `get_related_for_assets` body, therefore function signature is changed, the reasoning is below

Before my changes, there was a single inclusion of `asset_data` table in resulting sql query: `asset_data` was included in query because of joining to `asset` via `find_also_related(asset_data::Entity)` statement  in `get_assets_by_condition` function,  just before `get_related_for_assets` was called. 

Since my PR involved filtering by `asset_data.json_uri`, it lead to the inclusion of `asset_data` in filter query, that made impossible to use `find_also_related(asset_data::Entity)` statement  in `get_assets_by_condition` function, because it produced the error `Database Error Query Error: error returned from database: table name \"asset_data\" specified more than once"`.

To negotiate this, I've decided to remove the 2nd inclusion of `asset_data` in sql query by removing `find_also_related(asset_data::Entity)` in favour of explicit joining of `asset_data` to `asset` inside `get_related_for_assets()` function, therefore `get_related_for_assets` function signature is changed

@austbot, thoughts?

Reply:
This makes sense to me.  I see how it works.  Thank you for the explanation.